include/dlaf/matrix/tile.h

```diff
@@ -126,15 +141,48 @@ class Tile<const T, D>;
 
 namespace internal {
 template <class T, Device D>
-pika::shared_future<Tile<T, D>> splitTileInsertFutureInChain(pika::future<Tile<T, D>>& tile);
+using TileAsyncRwMutex =
+    pika::execution::experimental::async_rw_mutex<Tile<T, D>, const Tile<const T, D>>;
+
+template <class T, Device D>
+using TileAsyncRwMutexReadWriteWrapper = pika::execution::experimental::async_rw_mutex_access_wrapper<
+    Tile<T, D>, const Tile<const T, D>,
+    pika::execution::experimental::async_rw_mutex_access_type::readwrite>;
+
+template <class T, Device D>
+using TileAsyncRwMutexReadOnlyWrapper = pika::execution::experimental::async_rw_mutex_access_wrapper<
+    Tile<T, D>, const Tile<const T, D>, pika::execution::experimental::async_rw_mutex_access_type::read>;
+}
+
+template <class T, Device D>
+using ReadWriteTileSender = pika::execution::experimental::unique_any_sender<Tile<T, D>>;
+
+template <class T, Device D>
+using ReadOnlyTileSender =
+    pika::execution::experimental::any_sender<internal::TileAsyncRwMutexReadOnlyWrapper<T, D>>;
+
+namespace internal {
+// TODO: Do these need to be free functions or could they be hidden inside Tile?
+// They're already friends.
```

Comment:
No opinion on this right now. Just pointing in case you want to do something before merging 😉

Reply:
I've removed the TODO since I'm hopefully not making things worse by leaving them free functions. But eventually it'd be nice to do something about them...